lib/IRGen/GenPack.cpp

```diff
@@ -164,30 +164,9 @@ static Address emitFixedSizeMetadataPackRef(IRGenFunction &IGF,
   return pack;
 }
 
-static void emitPackExpansionType(IRGenFunction &IGF,
-                                  Address pack,
-                                  CanPackExpansionType expansionTy,
-                                  llvm::Value *dynamicIndex,
-                                  llvm::Value *dynamicLength,
-                                  DynamicMetadataRequest request) {
-  auto *prev = IGF.Builder.GetInsertBlock();
-  auto *check = IGF.createBasicBlock("pack-expansion-check");
-  auto *loop = IGF.createBasicBlock("pack-expansion-loop");
-  auto *rest = IGF.createBasicBlock("pack-expansion-rest");
-
-  IGF.Builder.CreateBr(check);
-  IGF.Builder.emitBlock(check);
-
-  // An index into the source metadata pack.
-  auto *phi = IGF.Builder.CreatePHI(IGF.IGM.SizeTy, 2);
-  phi->addIncoming(llvm::ConstantInt::get(IGF.IGM.SizeTy, 0), prev);
-
-  // If we reach the end, jump to the continuation block.
-  auto *cond = IGF.Builder.CreateICmpULT(phi, dynamicLength);
-  IGF.Builder.CreateCondBr(cond, loop, rest);
-
-  IGF.Builder.emitBlock(loop);
-
+static llvm::Value *emitPackExpansionElementMetadata(
+    IRGenFunction &IGF, CanPackExpansionType expansionTy, llvm::Value *phi,
```

Comment:
`phi` is an unfortunate name for a parameter, since you don't care whether it's a phi.  `index`, maybe?